CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -1063,6 +1294,77 @@ void SiStripGainFromCalibTree::algoComputeMPVandGain() {
 	}printf("\n");
 }
 
+void SiStripGainFromCalibTree::qualityMonitor() {
+
+    for(unsigned int a=0;a<APVsCollOrdered.size();a++) {
+        stAPVGain* APV = APVsCollOrdered[a];
+        if(APV==NULL)continue;
+
+        unsigned int  SubDet       = APV->SubDet;
+        float         z            = APV->z;
+        float         Eta          = APV->Eta;
+        float         R            = APV->R;
+        float         Phi          = APV->Phi;
+        float         Thickness    = APV->Thickness;
+        double        FitMPV       = APV->FitMPV;
+        double        FitMPVErr    = APV->FitMPVErr;
+        double        Gain         = APV->Gain;
+        double        NEntries     = APV->NEntries;
+        double        PreviousGain = APV->PreviousGain;
+
+        if(SubDet>=3 && FitMPV<0.) NoMPV->Fill(z,R);
+        if(SubDet>=3 && FitMPV>=0.) {
+            if(SubDet==3) MPV_Vs_EtaTIB->Fill(Eta,FitMPV);
+            if(SubDet==4) MPV_Vs_EtaTID->Fill(Eta,FitMPV);
+            if(SubDet==5) MPV_Vs_EtaTOB->Fill(Eta,FitMPV);
+            if(SubDet==6) MPV_Vs_EtaTEC->Fill(Eta,FitMPV);
+            if(SubDet==6 && Thickness<0.04) MPV_Vs_EtaTEC1->Fill(Eta,FitMPV);
+            if(SubDet==6 && Thickness>0.04) MPV_Vs_EtaTEC1->Fill(Eta,FitMPV);
```

Comment:
@dimattia, it appears there is a bug in these histograms in which the statistics should be separated according to the sensor thickness, i.e. you're filling twice the same monitor element.